docs/configuring.md

```diff
@@ -1035,6 +1036,151 @@ _commit: v1.0.0
 By default, copier will copy from the last release found in template git tags, sorted as
 [PEP 440][].
 
+## Augmenting Copier's context before rendering
+
+[TL;DR: Using a context hook](#using-a-context-hook)
+
+Most of the time, only a few questions/answers are needed to render simple templates.
+But as your template grows, both in complexity and functionality, you may find yourself
+in a situation where you need to ask more and more questions to users, and to use more
+and more Jinja conditions in your templated files, making them hard to read and
+maintain.
+
+As it happens, you find that to simplify your template, you would like to be able to add
+or change variables of the Jinja context used to render templated files and folders, as
+it would allow to get rid of some questions that would then become redundant, and would
+not need to be asked to the users anymore.
+
+An example was brought up by a Copier user. Let say you support multiple flavors of
+containerization for Terraform projects generated with your template:
+
+```yaml title="copier.yaml"
+flavor:
+    type: str
+    choices:
+        - Docker
+        - Instances
+        - Kubernetes
+        - None
+```
+
+You generate different files depending on users answers:
+
+```
+.
+├── {% if flavor == 'docker' %}ecs{% endif %}
+├── {% if flavor == 'kubernetes' %}eks{% endif %}
+├── {% if flavor == 'instance' %}bastion{% endif %}
+├── {% if flavor == 'docker' or flavor == 'kubernetes' %}portainer{% endif %}
+└── {% if flavor != 'none' %}vpc{% endif %}
+```
+
+Each of these folders then have multiple templated files that must also repeat again and
+again these Jinja conditions to render the contents correctly.
+
+You can reduce repetitions with
+[macros](https://jinja.palletsprojects.com/en/2.10.x/templates/#macros) or updated
+context objects that you can import from special templated files
+([see issue #229](https://github.com/copier-org/copier/issues/229#issuecomment-661686740)).
```

Comment:
I'd recommend using a [namespace](https://jinja.palletsprojects.com/en/3.1.x/templates/#jinja-globals.namespace) instead.

Reply:
Irrelevant now that this section is removed.